engine/pinger.go

```diff
@@ -98,5 +98,6 @@ func (p *pinger) ping(ctx context.Context, server *autoscaler.Server) error {
 
 	server.Error = "Failed to ping the server"
 	server.Stopped = time.Now().Unix()
+	server.State = autoscaler.StateStopped
```

Comment:
I think this should be `autoscaler.StateError` to indicate the server is in an unexpected state. You can use a feature flag to enable Drone to garbage collect errored servers.